### ISSUE TITLE
Addition to #34 : add matchExpressions

### ISSUE THREAD

**Issue by Tronix117** [reactiveops/rbac-manager#43]
Following the addition of matchLabels in #34, it could be great if it can also have the matchExpressions.

It can actualy be really tricky when you need to combine multiple labels which can take multiple values.

---

### Here a concrete exemple

```
criticality=danger, caution, none
environment=production, staging, qa, uat, development
tier=frontent, backend, cache, database
tenant=departmentA, departmentB, clientA
```

Now I'm in a situation similar to this one:
```
departmentA-admins
departmentA-ops
departmentA-devs

departmentB-admins
departmentB-ops
departmentB-devs

clientA-ops
clientA-devs
```

Some needs to have access to everything with a certain tenant/environment couple, so it's not perfect, but not this painful, for ex, `departmentA-ops` should have access to all environments except `production` on `frontend` tier, that is 4 rules. No if I through in that they can only access to it if criticality is not danger, then it jumps to 8 rules.

And that is not even the worst case I got :)

What if tomorrow I add a `beta` environment ? I need to add rules everywhere...

With `matchExpressions`, we only need 1 binding instead of 8 to achieve the same goal:
```
  matchExpressions:
    - {key: criticality, operator: NotIn, values: [danger]}
    - {key: environment, operator: NotIn, values: [production]}
```

**Comment by robscott** [comment]
Hey @Tronix117, I think that's a great idea. Completely agree that there are a number of use cases that would be helpful. We may be able to fit that into one of our next releases.

**Comment by robscott** [comment]
Hey @Tronix117 support for this feature just landed in 0.6.0, thanks for the great idea!
